.azuredevops/pipelineTemplates/jobs.validateModuleDeployment.yml

```diff
@@ -311,7 +311,7 @@ jobs:
       #------------------
       - task: AzurePowerShell@5
         displayName: 'Remove deployed resources via [${{ parameters.serviceConnection }}]'
-        condition: and(succeededOrFailed(), eq('${{ parameters.removeDeployment }}', 'True'), not(eq(variables['deploymentNames'],'')), not(startsWith(variables['deploymentNames'], 'variables[' )))
+        condition: and( always(), eq('${{ parameters.removeDeployment }}', 'True'), not(eq(variables['deploymentNames'],'')), not(startsWith(variables['deploymentNames'], 'variables[' )))
```

Comment:
I wonder, I wonder - isn't `succeededOrFailed()` better? Not being able to cancel a pipeline always gives me the creeps.

Honestly, no idea why it is not implemented this way on the GitHub-side (maybe it wasn't supported), but I'm inclined to say (if possible) we should rather align the GitHub condition with the Azure DevOps condition, rather than the other way around. 

Thoughts?

Reply:
I thought it made sense as you would then at least do the cleanup regardless.

Reply:
Right, you do want the cleanup - but isn't succeededOrFailed leading exactly to that? No matter if the deployment worked, the removal will run - unless - you cancel the pipeline. For example because you realized 'crap the deployment failed again, and I want to check it out in Azure before the pipeline tries to delete it'. Are we talking past each other? 👯

Reply:
Ye that does make sense! Lets close this PR.

Reply:
I'd be curious to see if we can achieve the same in GitHub by now (assuming it didn't work before). Will try to investigate :) Thanks @MariusStorhaug